### PR TITLE
Hotfix for NodeViewSynchronizer.parseModifiers

### PR DESCRIPTION
**Pull request by i-skrypnyk**
https://horizenlabs.atlassian.net/browse/SDK-471

### CODE REVIEW

File: src/main/scala/sparkz/core/network/NodeViewSynchronizer.scala [hunk 1]
```diff
@@ -325,13 +325,12 @@ class NodeViewSynchronizer[TX <: Transaction, SI <: SyncInfo, SIS <: SyncInfoMes
...
       val reader: VLQReader = new VLQByteBufferReader(ByteBuffer.wrap(bytes))
 
-      if (reader.remaining != 0) {
-        penalizeMisbehavingPeer(remote)
-        log.warn(s"Received additional bytes after block. Declared id ${encoder.encodeId(id)} from ${remote.toString}")
-      }
-
       serializer.parseTry(reader) match {
         case Success(mod) if id == mod.id =>
+          if (reader.remaining != 0) {
```
Comment (i-Alex) [inline src/main/scala/sparkz/core/network/NodeViewSynchronizer.scala:330]: Let's have some UTs on Sparkz level. Don't want to keep it untested within the Sparkz project, but only on SDK side.
Note: you may define a dummy tx or block serializers for this.

**Review by i-Alex — CHANGES_REQUESTED** [review]

File: src/main/scala/sparkz/core/network/NodeViewSynchronizer.scala (hunk 1 above)
Comment (i-skrypnyk) [reply src/main/scala/sparkz/core/network/NodeViewSynchronizer.scala:330]: Added UT covering these cases

**Review by i-Alex — APPROVED** [review]

File: src/test/scala/sparkz/core/network/NodeViewSynchronizerSpecification.scala [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    viewHolder.expectMsg(TransactionsFromRemote(Seq(transaction)))
+    networkController.expectMsg(PenalizePeer(peer.connectionId.remoteAddress, MisbehaviorPenalty))
+  }
+
+  def roundTrip(msg: Message[_]): Message[_] = {
+    messageSerializer.deserialize(messageSerializer.serialize(msg), msg.source) match {
+      case Failure(e) => throw e
+      case Success(value) => value match {
+        case Some(value) => value
+        case None => fail("")
```
Comment (paologalligit) [inline src/test/scala/sparkz/core/network/NodeViewSynchronizerSpecification.scala]: Let's fill the empty string with something like "roundTrip test method unexpected behavior"

**Review by paologalligit — APPROVED** [review]
LGTM! Just minor comment

File: src/test/scala/sparkz/core/network/NodeViewSynchronizerSpecification.scala (hunk 2 above)
Comment (i-skrypnyk) [reply src/test/scala/sparkz/core/network/NodeViewSynchronizerSpecification.scala]: thanks, fixed

**Pull request merged**